Index: configs/modules.conf.sample
===================================================================
--- configs/modules.conf.sample	(revision 216642)
+++ configs/modules.conf.sample	(working copy)
@@ -38,3 +38,29 @@
 ;noload => chan_oss.so
 noload => chan_console.so
 ;
+
+;
+; Voicemail storage selection
+;
+; Comment out the "noload" lines for the voicemail
+; storage system that you want.  Leave the ones that
+; you don't want uncommented.
+;
+
+;
+; Voicemail with IMAP storage
+;
+noload => app_directory_imap.so
+noload => app_voicemail_imap.so
+
+;
+; Voicemail with ODBC storage
+;
+noload => app_directory_odbc.so
+noload => app_voicemail_odbc.so
+
+;
+; Voicemail with filesystem storage
+;
+;noload => app_directory_plain.so
+;noload => app_voicemail_plain.so
