From c585c45d9e9d2ef30bd1db15454ef707fc22559b Mon Sep 17 00:00:00 2001
From: Jeffrey C. Ollie <jeff@ocjtech.us>
Date: Wed, 28 Jul 2010 07:23:49 -0500
Subject: [PATCH 2/8] Modify modules.conf so that different voicemail modules don't load at the same time.

---
 configs/modules.conf.sample |   26 ++++++++++++++++++++++++++
 1 files changed, 26 insertions(+), 0 deletions(-)

diff --git a/configs/modules.conf.sample b/configs/modules.conf.sample
index 5b553a4..39645c3 100644
--- a/configs/modules.conf.sample
+++ b/configs/modules.conf.sample
@@ -35,3 +35,29 @@ noload => chan_alsa.so
 ;noload => chan_oss.so
 noload => chan_console.so
 ;
+
+;
+; Voicemail storage selection
+;
+; Comment out the "noload" lines for the voicemail
+; storage system that you want.  Leave the ones that
+; you don't want uncommented.
+;
+
+;
+; Voicemail with IMAP storage
+;
+noload => app_directory_imap.so
+noload => app_voicemail_imap.so
+
+;
+; Voicemail with ODBC storage
+;
+noload => app_directory_odbc.so
+noload => app_voicemail_odbc.so
+
+;
+; Voicemail with filesystem storage
+;
+;noload => app_directory_plain.so
+;noload => app_voicemail_plain.so
-- 
1.7.2.1

