From e5d3f951bfa543e5d8b8cf85535f80b1d1143e09 Mon Sep 17 00:00:00 2001
From: Jeffrey C. Ollie <jcollie@pc21224.campus.dmacc.edu>
Date: Thu, 8 Nov 2007 15:52:36 -0600
Subject: [PATCH 02/11] Modify modules.conf so that different voicemail modules can be loaded.

---
 configs/modules.conf.sample |   26 ++++++++++++++++++++++++++
 1 files changed, 26 insertions(+), 0 deletions(-)

diff --git a/configs/modules.conf.sample b/configs/modules.conf.sample
index 8f73de1..7dd8940 100644
--- a/configs/modules.conf.sample
+++ b/configs/modules.conf.sample
@@ -44,3 +44,29 @@ noload => chan_alsa.so
 ; 
 ;noload => res_timing_dahdi.so
 ;noload => res_timing_pthread.so
+
+;
+; Voicemail storage selection
+;
+; Comment out the "noload" lines for the voicemail
+; storage system that you want.  Leave the ones that
+; you don't want uncommented.
+;
+
+;
+; Voicemail with IMAP storage
+;
+noload => app_directory_imap.so
+noload => app_voicemail_imap.so
+
+;
+; Voicemail with ODBC storage
+;
+noload => app_directory_odbc.so
+noload => app_voicemail_odbc.so
+
+;
+; Voicemail with filesystem storage
+;
+;noload => app_directory_plain.so
+;noload => app_voicemail_plain.so
-- 
1.6.1

