From 635e74de63418f0cd61ec5b1098e451a17b7d360 Mon Sep 17 00:00:00 2001
From: Jeffrey C. Ollie <jeff@ocjtech.us>
Date: Fri, 7 Nov 2008 14:43:16 -0600
Subject: [PATCH 07/15] Define missing variable when compiling on PPC.

---
 apps/app_festival.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/apps/app_festival.c b/apps/app_festival.c
index 649707b..34c282b 100644
--- a/apps/app_festival.c
+++ b/apps/app_festival.c
@@ -119,6 +119,7 @@ static int send_waveform_to_fd(char *waveform, int length, int fd)
 {
 	int res;
 #ifdef __PPC__ 
+	int x;
 	char c;
 #endif
 
-- 
1.6.1

