From a092080315b7bfa9b7d174b6114cb957ac1590d8 Mon Sep 17 00:00:00 2001
From: Jeffrey C. Ollie <jeff@ocjtech.us>
Date: Fri, 10 Oct 2008 11:54:10 -0500
Subject: [PATCH 08/15] Revert changes to pbx_lua from rev 126363 that cause some problems.

---
 pbx/pbx_lua.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/pbx/pbx_lua.c b/pbx/pbx_lua.c
index 6a600a4..a803bd5 100644
--- a/pbx/pbx_lua.c
+++ b/pbx/pbx_lua.c
@@ -42,9 +42,9 @@ ASTERISK_FILE_VERSION(__FILE__, "$Revision$")
 #include "asterisk/paths.h"
 #include "asterisk/hashtab.h"
 
-#include <lua5.1/lua.h>
-#include <lua5.1/lauxlib.h>
-#include <lua5.1/lualib.h>
+#include <lua.h>
+#include <lauxlib.h>
+#include <lualib.h>
 
 static char *config = "extensions.lua";
 static char *registrar = "pbx_lua";
-- 
1.6.1

