Index: res/res_http_post.c
===================================================================
--- res/res_http_post.c	(revision 221084)
+++ res/res_http_post.c	(working copy)
@@ -122,14 +122,10 @@
 		ast_log(LOG_WARNING, "Got unexpected GMIME_IS_MESSAGE_PARTIAL\n");
 		return;
 	} else if (GMIME_IS_MULTIPART(part)) {
-		GList *l;
-		
 		ast_log(LOG_WARNING, "Got unexpected GMIME_IS_MULTIPART, trying to process subparts\n");
-		l = GMIME_MULTIPART(part)->subparts;
-		while (l) {
-			process_message_callback(l->data, cbinfo);
-			l = l->next;
-		}
+
+		g_mime_multipart_foreach(GMIME_MULTIPART(part), process_message_callback,
+                               cbinfo);
 	} else if (GMIME_IS_PART(part)) {
 		const char *filename;
 
@@ -151,7 +147,7 @@
 		.post_dir = post_dir,
 	};
 
-	g_mime_message_foreach_part(message, process_message_callback, &cbinfo);
+	g_mime_message_foreach(message, process_message_callback, &cbinfo);
 
 	return cbinfo.count;
 }
