Index: res/res_http_post.c
===================================================================
--- res/res_http_post.c	(revision 226100)
+++ res/res_http_post.c	(working copy)
@@ -447,7 +447,7 @@
 
 			urih->description = ast_strdup("HTTP POST mapping");
 			urih->uri = ast_strdup(v->name);
-			ast_str_set(&ds, 0, "%s/%s", prefix, v->value);
+			ast_str_set(&ds, 0, "%s", v->value);
 			urih->data = ds;
 			urih->has_subtree = 0;
 			urih->supports_get = 0;
