--- a/res/Makefile	2022-04-07 16:31:40.365791898 +0200
+++ b/res/Makefile	2022-04-07 16:31:31.592727512 +0200
@@ -74,6 +74,8 @@
 $(call MOD_ADD_C,res_stir_shaken,$(wildcard res_stir_shaken/*.c))
 
 res_parking.o: _ASTCFLAGS+=$(AST_NO_FORMAT_TRUNCATION)
+snmp/agent.o: _ASTCFLAGS+=-fPIC
+res_snmp.o: _ASTCFLAGS+=-fPIC
 
 # Dependencies for res_ari_*.so are generated, so they're in this file
 include ari.make
